d2/src/test/java/com/linkedin/d2/balancer/servers/ZooKeeperServerTest.java

```diff
@@ -184,6 +192,13 @@ public void testZkServer() throws Exception
     assertEquals(properties.getUriSpecificProperties().size(), 1);
     assertEquals(properties.getUriSpecificProperties().get(URI_2), _uri2SpecificProperties);
 
+    UriProperties propertiesForUri2 = new UriProperties(properties.getClusterName(),
```

Comment:
When we announce, we only have access to the properties for *this URI specifically* which will normally be different from the entire properties we load from D2 (as this will contain data for all URIs in the cluster)

One consequence of this is that whichever entity _consumes_ the announcements and deannouncements will need to also manage merging the `UriProperties` for a cluster from the fragments passed by each announcement

Reply:
Sounds good, glad to hear this won't be an issue!